### PR TITLE
Make network version a variable to reduce recompilations

### PR DESCRIPTION
**Pull request by janisozaur**
Touching `network.h` requires whole lot of needless recompilation, moving network version to the implementation make changing the network version one file compilation and relinking.

### CODE REVIEW

File: src/openrct2/network/Network.cpp [hunk 1]
```diff
@@ -1018,7 +1024,7 @@ void Network::Client_Send_AUTH(const char* name, const char* password, const cha
 {
     std::unique_ptr<NetworkPacket> packet(NetworkPacket::Allocate());
     *packet << (uint32)NETWORK_COMMAND_AUTH;
-    packet->WriteString(NETWORK_STREAM_ID);
+    packet->WriteString(network_get_version().c_str());
```
Comment (Gymnasiast) [inline src/openrct2/network/Network.cpp]: Why is this necessary? Can't you keep using NETWORK_STREAM_ID here?

Comment (janisozaur) [reply src/openrct2/network/Network.cpp]: I'd rather expose and use a single interface everywhere

Comment (Gymnasiast) [reply src/openrct2/network/Network.cpp]: Oh, alright then. This will need to be rebased though, and the network version has increased in the meantime (from 31 to 32).

Comment (janisozaur) [reply src/openrct2/network/Network.cpp]: > Edit: rebased it.

please don't edit the comment to say so, make another one. It's easy to miss such an edit.
